src/libse/Common/StringExtensions.cs

```diff
@@ -756,6 +756,46 @@ public static bool HasSentenceEnding(this string value)
             return value.HasSentenceEnding(string.Empty);
         }
 
+        /// <summary>
+        /// Evaluates whether all of the letter-characters in the given string are uppercase.
+        /// </summary>
+        /// <param name="value">The string to evaluate.</param>
+        /// <remarks>
+        /// If the string is null or empty, the method will return false.
+        /// The method only checks letter-characters and ignores any non-letter characters.
+        /// If the string doesn't contain any letters, the method returns false.
+        /// </remarks>
+        /// <returns>
+        /// Returns true if all letter-characters in the string are uppercase. Returns false otherwise.
+        /// </returns>
+        public static bool IsAllUppercase(this string value)
```

Comment:
The documentation explain very well when to call this method to check if string is all uppercase!

One of the thing not to do is to call this method and try use its inverse value to say string is lowercase, 
the reason is that is *vaule* is null it will return `false` but that doesn't mean the string is all lowercase!

If we need `IsAllLowercase` it's better to create it basing on what it's done for `IsAllUppercase`